File: 217. Contains Duplicate/Approach_1.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(),nums.end());
        for (int i = 0; i < n-1; i++) {
            int j = i + 1;
            if(nums[i] == nums[j]){
                return true;
            }
        }
        return false;
    }
};

File: 4038. Count Integers Appearing in a Single Block/Approach_1.java
class Solution {
    private boolean Checkforconsecutive(List<Integer> indxarr) {
        for (int i = 0; i < indxarr.size() - 1; i++) {
            if (indxarr.get(i+1) - indxarr.get(i) > 1)
                return false;
        }
        return true;
    }

    public int countSpecialIntegers(int[] nums) {
       
        int n = nums.length;
        Map<Integer, List<Integer>> mymap = new HashMap<>();

        int cnt = 0;
        for (int i = 0; i < n; i++) {
            if (!mymap.containsKey(nums[i])) {
                mymap.put(nums[i], new ArrayList<>());
            }
            mymap.get(nums[i]).add(i);
        }

        for (Map.Entry<Integer, List<Integer>> e : mymap.entrySet()) {
            List<Integer> indxarr = e.getValue();
            if (Checkforconsecutive(indxarr) == true) {
                cnt++;
            }
        }
        return cnt;
    }
}